controllers/cloud.redhat.com/namespacereservation_controller.go

```diff
@@ -96,6 +96,8 @@ func (r *NamespaceReservationReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 		if r.Poller.namespaceIsExpired(expirationTS) {
+			DeletePrometheusOperator(ctx, r.Client, res.Status.Namespace)
```

Comment:
DeletePrometheusOperator returns an error but this is not caught here: dragons will rear their heads!